test/src/test/java/aima/test/unit/search/uninformed/BidirectionalSearchTest.java

```diff
@@ -29,50 +40,57 @@
     @Parameterized.Parameter
     public String searchFunctionName;
 
-    public <A, S> BidirectionalActions<A> searchForActions(Problem<A, S> originalProblem, Problem<A, S> reverseProblem) {
-        BidirectionalSearch<A, S> searchForActionsFunction = new BidirectionalSearch<A, S>();
-        return searchForActionsFunction.apply(originalProblem, reverseProblem);
+    public <A, S> BidirectionalActions<A> searchForActions(Pair<Problem<A, S>, Problem<A, S>> pair) {
```

Comment:
https://github.com/aimacode/aima-java/pull/294#issuecomment-290656910
> Primary purpose of BidirectionalActions is to enhance the functionality of the search algorithm. It adds functionality to the API.

The name should convey it's purpose though http://www.itiseezee.com/?p=83
everything else makes it harder for everyone trying to understand the code